packages/@lwc/babel-plugin-component/src/dynamic-imports.ts

```diff
@@ -1,45 +1,52 @@
 /*
- * Copyright (c) 2018, salesforce.com, inc.
+ * Copyright (c) 2023, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-const moduleImports = require('@babel/helper-module-imports');
-const { LWCClassErrors } = require('@lwc/errors');
+import { NodePath } from '@babel/traverse';
+import { types, Visitor } from '@babel/core';
+import { addNamed } from '@babel/helper-module-imports';
+import { LWCClassErrors } from '@lwc/errors';
+import { generateError } from './utils';
+import { LwcBabelPluginPass } from './types';
 
-const { generateError } = require('./utils');
-
-function getImportSource(path) {
-    return path.parentPath.get('arguments.0');
+function getImportSource(path: NodePath<types.Import>): NodePath<types.Node> {
+    return path.parentPath.get('arguments.0') as NodePath<types.Node>;
 }
 
-function validateImport(sourcePath) {
+function validateImport(sourcePath: NodePath<types.Node>) {
     if (!sourcePath.isStringLiteral()) {
         throw generateError(sourcePath, {
             errorInfo: LWCClassErrors.INVALID_DYNAMIC_IMPORT_SOURCE_STRICT,
             messageArgs: [String(sourcePath)],
         });
     }
 }
+
 /*
  * Expected API for this plugin:
  * { dynamicImports: { loader: string, strictSpecifier: boolean } }
  */
-module.exports = function () {
-    function getLoaderRef(path, loaderName, state) {
+export default function (): Visitor<LwcBabelPluginPass> {
+    function getLoaderRef(
+        path: NodePath<types.Import>,
+        loaderName: string,
+        state: LwcBabelPluginPass
+    ): types.Identifier {
         if (!state.loaderRef) {
-            state.loaderRef = moduleImports.addNamed(path, 'load', loaderName);
+            state.loaderRef = addNamed(path, 'load', loaderName);
         }
         return state.loaderRef;
     }
 
-    function addDynamicImportDependency(dependency, state) {
+    function addDynamicImportDependency(dependency: string, state: LwcBabelPluginPass) {
         if (!state.dynamicImports) {
             state.dynamicImports = [];
         }
 
         if (!state.dynamicImports.includes(dependency)) {
-            state.dynamicImports.push(dependency);
+            state.dynamicImports!.push(dependency);
```

Comment:
Where is this consumed? I can't any reference to this `dynamicImports`.

Reply:
Correct, it seems unused. Maybe this is a mistake because they meant to put it on the `options.dynamicImports` instead?

Reply:
Added a TODO.